nncf/experimental/common/tensor_statistics/statistical_functions.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nncf.experimental.tensor import Tensor
+from nncf.experimental.tensor import functions as fns
+
+
+def mean_per_channel(x: Tensor, axis: int) -> Tensor:
+    """
+    Computes the mean of elements across given channel dimension of Tensor.
+
+    :param x: Tensor to reduce.
+    :param axis: The channel dimensions to reduce.
+    :return: Reduced Tensor.
+    """
+    if len(x.shape) < 3:
+        return fns.mean(x, axis=0)
+    x = fns.moveaxis(x, axis, 1)
+    t = x.reshape([x.shape[0], x.shape[1], -1])
+    return fns.mean(t, axis=(0, 2))
```

Comment:
You can do this in a shorter fashion:
```python
axis = tuple(i for i in range(x.ndim) if i != channel_axis)
return fns.mean(t, axis=axis)
```

Reply:
```python
    pos_axis = axis + x.ndim if axis < 0 else axis
    if not 0 <= pos_axis < x.ndim:
        raise ValueError(f"axis {axis} is out of bounds for array of dimension {x.ndim}")
    axis = tuple(i for i in range(x.ndim) if i != pos_axis)
    return fns.mean(x, axis=axis)
```
Not shorter, but it looks clearer 👍 


About performance, looks the same in most cases but in case of `shape: 1, 256, 16, 16` and `axis=1` speedup = `1.18`
```

    x = np.random.rand(1, i, 256, 256)
    axis = 1
-------
Shape: 1, 16, 256, 256
New: 3.24s  Old: 3.3s  Speedup: 1.0177
-------
Shape: 1, 48, 256, 256
New: 12.8s  Old: 12.8s  Speedup: 0.99951
-------
Shape: 1, 96, 256, 256
New: 31.8s  Old: 31.5s  Speedup: 0.99251
-------
Shape: 1, 256, 256, 256
New: 94.9s  Old: 94.9s  Speedup: 1.0

    x = np.random.rand(1, 256, i, i)
    axis = 1
-------
Shape: 1, 256, 16, 16
New: 0.283s  Old: 0.335s  Speedup: 1.1816
-------
Shape: 1, 256, 48, 48
New: 2.02s  Old: 2.09s  Speedup: 1.0323
-------
Shape: 1, 256, 96, 96
New: 9.57s  Old: 9.63s  Speedup: 1.0058
-------
Shape: 1, 256, 256, 256
New: 95.7s  Old: 95.8s  Speedup: 1.0009
```
I will update after https://github.com/openvinotoolkit/nncf/pull/2058#discussion_r1322651908

Reply:
updated